content/docs/guides/testing/_index.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Testing
+meta_desc: "Guide to automated testing of Pulumi programs: unit testing, integration testing, Policy as Code."
+menu:
+    userguides:
+        identifier: testing
+        weight: 7
+---
+
+Pulumi uses general-purpose programming languages to define cloud infrastructure. This capability opens up all the doors to **automated testing** of the infrastructure resources.
+
+The software industry has come up with multiple styles of automated testing. Unit testing of individual pieces, integration testing of system components, end-to-end testing of the complete application&mdash;each style has its advantages and challenges.
+
+Pulumi provides a similar choice of testing options for cloud programs. Currently, we identify three types of tests:
+
+- **Unit Tests** are fast in-memory tests that mock all external calls.
+- **Tests based on Policy as Code** run resource-level assertions *while* infrastructure is being deployed.
+- **Integration Tests** deploy ephemeral infrastructure and run external tests against it.
+
+The following table summarizes the differences between the three approaches:
+
+|                                | Unit Tests  | Policy as Code  | Integration Tests  |
+|--------------------------------|-----|------|------|
+| Provision real infrastructure  | No  | Yes  | Yes  |
+| Require the Pulumi CLI         | No  | Yes  | Yes  |
+| Time to execute                | Milliseconds  | Seconds to minutes  | Seconds to minutes  |
+| Language                       | Same as Pulumi program  | Node.js or Python  | Any language  |
```

Comment:
We don't yet have bespoke libraries for "any language" for Integration testing, but depending on how you frame it in the details, this seems fine as is (that is, if you position as a workflow of doing `pulumi up`, "run validation" and `pulumi destroy` that can be scripted in any language).

Reply:
Yes, I'll describe this in the details